### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by serg121192**

### CODE REVIEW

**Review by Abnormaltype — CHANGES_REQUESTED** [review]
you can use args as the key and the result of the function as a value. Try to do it

**Comment by serg121192** [comment]
> you can use args as the key and the result of the function as a value. Try to do it
I'm sorry, but I thought I'd already realized this idea in my code, didn't it? I'm about this string: _cache[func.__name__][inputs] = func(*inputs). A dictionary _cache, where stored the names of decorated functions as keys of dictionaries, and theirs values are the dictionaries too, where the keys are the args, and values - the results of functions. Or I didn't understand what's your idea. please, explain :)

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
-    pass
+from typing import Any
+
+
+def cache(func: callable) -> callable:
+    _cache = {}
+
+    def some_inputs(*inputs: Any) -> str:
+        if func.__name__ not in _cache:
+            _cache[func.__name__] = {}
```
Comment (Abnormaltype) [inline app/main.py:9]: you store **name of the function** -> you should **args**

Comment (Abnormaltype) [reply app/main.py:9]: u use dict in dict - it's unnecessary

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (serg121192) [reply app/main.py:9]: > u use dict in dict - it's unnecessary

okay, I'll fix it. thx)

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any
+
+
+def cache(func: callable) -> callable:
+    _cache = {}
+
+    def some_inputs(*inputs: Any) -> Any:
```
Comment (Vasyl-Smutok) [inline app/main.py:7]: 1. I would rename `some_inputs` to `wrapper`
2. *inputs: tuple

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any
+
+
+def cache(func: callable) -> callable:
+    _cache = {}
+
```
Comment (Vasyl-Smutok) [inline app/main.py:6]: added     `@wraps(func)`

**Review by vsmutok — CHANGES_REQUESTED** [review]

**Review by vsmutok — APPROVED** [review]
Well done 👍

**Pull request closed without merging**